modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/LogicalRelImplementor.java

```diff
@@ -1158,4 +1144,20 @@ private static boolean canFuseProjectionInto(RelNode rel) {
 
         return joinProjection;
     }
+
+    private long validateAndGetFetchOffsetParams(RexNode node, String op) {
```

Comment:
probably we can move this check into Node code and append additional tests if it helpful with another issue?